cli/src/rust_template.rs

```diff
@@ -171,7 +174,17 @@ codegen-units = 1
 "#
 }
 
-fn cargo_toml(name: &str) -> String {
+fn cargo_toml(name: &str, with_mollusk: bool) -> String {
+    let test_sbf_feature = if with_mollusk { r#"test-sbf = []"# } else { "" };
+    let dev_dependencies = if with_mollusk {
+        r#"
+[dev-dependencies]
+mollusk-svm = "=0.0.6-solana-1.18"
```

Comment:
Any specific reason to use fixed versions? We're also on Solana v2 now (I see `mollusk-svm 0.0.7-solana-2.0` available), but I guess it doesn't matter much.

This is not a blocker for getting this in, but please let me me know if we can relax this requirement to avoid repetitive bumps.

Reply:
Ohh, maybe I wasn't understanding how the versioning worked with `master`. It seemed like the tip was generating a template for Anchor 0.30, which should be running 1.18, right? My lockfile was pulling in two versions when I used a version of Mollusk compatible with >2.0.

But yeah, there's a 2.0-compat release line as well as the main release line which supports 2.1.

Reply:
`anchor init` creates templates with `anchor-lang 0.30.1` because that's what the current `master` version is. However, `master` branch is already on Solana v2.1, so when we bump the version to v0.31.0 (https://github.com/coral-xyz/anchor/pull/3259), it will use the latest Solana version of v2.

In short, the `init` command is not up-to-date with `master`, as it always uses the last released version rather than directly depending on the `master` branch.

Reply:
So does that mean we'd adjust the version for `mollusk-svm` after the next Anchor release? Because trying to do it now causes the test job to fail.
https://github.com/coral-xyz/anchor/actions/runs/11814814004/job/32914710753?pr=3361

Reply:
Don't think that would make sense, since people would have the incompatible version with the new release that way.

The `init` test is a bit weird because it's testing the past release compared to all other tests using the latest `master`. It's probably best to bump `mollusk-svm` version during Anchor version bumps e.g. in https://github.com/coral-xyz/anchor/pull/3259. We could also bump it beforehand, but then we'd have CI failures until the next version is released.

Reply:
Ahh cool, yeah that makes sense to me. Do you want to add it to #3259? Version `0.0.8` is compatible with Solana `2.1`.

Reply:
Yep, will add it to https://github.com/coral-xyz/anchor/pull/3259